### ISSUE TITLE
Group api improvment

### ISSUE THREAD

**Issue by jflefebvre06** [gitlab4j/gitlab4j-api#482]
Hello,

Coud you please add the belowed variables to group API (create and update) :

- membership_lock (allowed only into deprecated method)
- share_with_group_lock (allowed only into deprecated method)
- project_creation_level
- subgroup_creation_level
- auto_devops_enabled

**Comment by gmessner** [comment]
@jflefebvre06 
GitLab4J-API 4.14.2 has been released with new group create and update methods that take a ```GroupParams``` instance.  Here is an example use:
```
GroupParams mods = new GroupParams()
    .withProjectCreationLevel(AccessLevel.NONE)
    .withAutoDevopsEnabled(true);
gitLabApi.getGroupApi().updateGroup(myGroupId, mods);
```

When you have time please verify and close this issue.

**Comment by jflefebvre06** [comment]
Hello Greg, thanks a lot

**Comment by jflefebvre06** [comment]
Hello greg,
 .withProjectCreationLevel(AccessLevel.MAINTAINER)
.withSubgroupCreationLevel(AccessLevel.OWNER)

does not work, ids must be replaced by string  `noone`, `maintainer` ...

**Comment by gmessner** [comment]
@jflefebvre06 
This has been fixed and release 4.14.3 is now available.

**Comment by jflefebvre06** [comment]
.withProjectCreationLevel(AccessLevel.MAINTAINER) works fine but .withSubgroupCreationLevel(AccessLevel.OWNER) failed "Invalid access level"

Missing case into method 

```java
private String getAccessLevelString(AccessLevel level) {

	if (level == null) {
	    return null;
	}

	switch(level) {

	case NONE:
	    return "noone";

	case DEVELOPER:
	    return "developer";

	case MAINTAINER:
	    return "maintainer";

	default:
	    throw new RuntimeException("Invalid access level");
	}
    }
```

**Comment by jflefebvre06** [comment]
I Shing you muist crete 2 methods getAccessLevelString, one for withProjectCreationLevel and another one for withSubgroupCreationLevel

**Comment by jflefebvre06** [comment]
Suggestion, replace `throw new RuntimeException("Invalid access level");` by `IllegalArgumentException`

**Comment by gmessner** [comment]
@jflefebvre06 
Actually, using the ```AccessLevel``` enum was probably not a good idea, so I have modified the ```GroupParams``` class to use newly created enums named ```ProjectCreationLevel``` and ```SubgroupCreationLevel```.

I will be doing a release which includes this change in the next day or so.

**Comment by jflefebvre06** [comment]
Ok thank you.

**Comment by gmessner** [comment]
@jflefebvre06 
Just published release 4.14.4 with this mod in it.

**Comment by jflefebvre06** [comment]
Tested, works fine !
